File: Abstract Factory/PHP/TeamPromoConnectionFactory.php
<?php

require_once "IConnectionFactory.php";
require_once "TeamPromoRPCConnection.php";
require_once "TeamPromoHTTPConnection.php";

class TeamPromoConnectionFactory implements IConnectionFactory {
    public function createRPCConnection() {
        return new TeamPromoRPCConnection;
    }

    public function createHTTPConnection() {
        return new TeamPromoHTTPConnection;
    }
}

File: Abstract Factory/Python 3/TeamPromoConnectionFactory.py
from IConnectionFactory import IConnectionFactory
from TeamPromoHTTPConnection import TeamPromoHTTPConnection
from TeamPromoRPCConnection import TeamPromoRPCConnection

class TeamPromoConnectionFactory(IConnectionFactory):
  def createHTTPConnection(self):
    return TeamPromoHTTPConnection()

  def createRPCConnection(self):
    return TeamPromoRPCConnection()